File: fixtures/scale/julia/large/src/core/Log.jl
module BenchLog

using Dates

export log_write, log_info, log_warn

function log_write(level::String, parts...)
    stamp = Dates.format(Dates.now(), "HH:MM:SS")
    println(stderr, stamp, " [", level, "] ", join(parts, " "))
end

function log_info(parts...)
    log_write("INFO", parts...)
end

function log_warn(parts...)
    log_write("WARN", parts...)
end

end

File: fixtures/scale/r/large/core/log.r
# Timestamped logging for bench runs.

log_write <- function(level, ...) {
  stamp <- format(Sys.time(), "%H:%M:%S")
  message(sprintf("%s [%s] %s", stamp, level, paste(..., sep = " ")))
}

log_info <- function(...) {
  log_write("INFO", ...)
}

log_warn <- function(...) {
  log_write("WARN", ...)
}
